tests/BenchmarkDotNet.IntegrationTests/MultipleInProcessDiagnosersTests.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Columns;
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.IntegrationTests.Diagnosers;
+using BenchmarkDotNet.Jobs;
+using BenchmarkDotNet.Tests.Loggers;
+using BenchmarkDotNet.Toolchains.InProcess.Emit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace BenchmarkDotNet.IntegrationTests;
+
+public class MultipleInProcessDiagnosersTests : BenchmarkTestExecutor
+{
+    public MultipleInProcessDiagnosersTests(ITestOutputHelper output) : base(output) { }
+
+    public static IEnumerable<object[]> GetDiagnoserCombinations()
+    {
+        // Two diagnosers with NoOverhead
```

Comment:
Use loops to test all combination of run modes for each count of diagnosers.
Also test each toolchain - InProcessEmitToolchain, InProcessNoEmitToolchain, and default toolchain.